examples/oneapi/dpc/CMakeLists.txt

```diff
@@ -16,7 +16,6 @@
 
 cmake_minimum_required(VERSION 3.1)
 
-set(CMAKE_BUILD_TYPE Release)
```

Comment:
Should be provided via command line `-DCMAKE_BUILD_TYPE={Release,Debug}`

Reply:
Is it really necessary to remove it, because, for me, looks that with `set(CMAKE_BUILD_TYPE Release)` it will be set by default, and from command line you can -DCMAKE_BUILD_TYPE=Debug

Reply:
For me, this line overrode what I specified with `-DCMAKE_BUILD_TYPE`. That's why I removed it. With this line, the link flag was not added correctly.

Reply:
What do you think about

```
if(NOT CMAKE_BUILD_TYPE)
    set(CMAKE_BUILD_TYPE Release CACHE STRING "Build type not specified, defaulting to Release" FORCE)
endif()
```

Reply:
I think that its worse to remove default value, as a basic example:
if(NOT CMAKE_BUILD_TYPE)
    set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build (Debug or Release)")
endif()

Reply:
but it works for same syntax for DPCPP/oneDAL vars below? 

I think that you have problems with cashing your variables and clean runs wouldn't be impacted

https://cmake.org/cmake/help/book/mastering-cmake/chapter/CMake%20Cache.html

Reply:
Don't think caching can be a factor here, as I deleted the build directory between every run. I can demonstrate

```
# WITH set(CMAKE_BUILD_TYPE Release) in place
$ rm -rf Build/ && mkdir Build
$ CC=icx CXX=icpx cmake -B./Build -S. -G "Unix Makefiles" -DONEDAL_LINK=static -DCMAKE_BUILD_TYPE=Debug
# ...
$ grep -r flink-huge-device-code Build/ | wc -l
0

# WITHOUT set(CMAKE_BUILD_TYPE Release) (original version from this PR)
$ rm -rf Build/ && mkdir Build
$ CC=icx CXX=icpx cmake -B./Build -S. -G "Unix Makefiles" -DONEDAL_LINK=static -DCMAKE_BUILD_TYPE=Debug
# ...
$ grep -r flink-huge-device-code Build/ | wc -l                                                                                                                                                                                                                                                                                                                                                                                                                     
30 

# WITH the if() statement (current version of this PR)
$ rm -rf Build/ && mkdir Build
$ CC=icx CXX=icpx cmake -B./Build -S. -G "Unix Makefiles" -DONEDAL_LINK=static -DCMAKE_BUILD_TYPE=Debug
# ...
$ grep -r flink-huge-device-code Build/ | wc -l                                                                                                                                                                                                                                                                                                                                                                                                                     
30 

# For completeness, without -DCMAKE_BUILD_TYPE to illustrate defaul behavior
$ rm -rf Build/ && mkdir Build
$ CC=icx CXX=icpx cmake -B./Build -S. -G "Unix Makefiles" -DONEDAL_LINK=static 
# ...
$ grep -r flink-huge-device-code Build/ | wc -l                                                                                                                                                                                                                                                                                                                                                                                                                     
0 
```

Reply:
I think it then worth moving this to https://github.com/oneapi-src/oneDAL/blob/master/cmake/templates/oneDALConfig.cmake.in

So we would handle this param passing for all cases, not only for DPC examples.